doc/src/content/xdocs/gettingstartedjava.xml

```diff
@@ -288,7 +288,13 @@ System.out.println(user);
           class, in this case <code>User</code>.  We pass the
           <code>DatumReader</code> and the previously created <code>File</code>
           to a <code>DataFileReader</code>, analogous to the
-          <code>DataFileWriter</code>, which reads the data file on disk.
+          <code>DataFileWriter</code>, which reads both the schema used by the
+          writer as well as the data from the file on disk. The data will be
+          read using the schema included in the file, and converted to the
```

Comment:
Could you remove the wording here about reading using the writer's schema and converting? Instead, it should be something like "the data will be read using the schema included in the file and the schema provided by the reader, in this case the User class. The writer's schema is needed to know the order in which fields were written, while the reader schema is needed to know what fields are expected and how to fill in default values for fields added since the file was written." Maybe that helps to make it more clear?